extensions/interactions/NumberWithUnits/directives/number-with-units-rules.service.ts

```diff
@@ -47,14 +47,13 @@ export class NumberWithUnitsRulesService {
     var answerObject = this.unitsObjectFactory.fromDict(answer);
     var inputsObject = this.unitsObjectFactory.fromDict(inputs.f);
 
-    var answerString = answerObject.toMathjsCompatibleString();
-    var inputsString = inputsObject.toMathjsCompatibleString();
+    var answerNum = answerObject.toMathjsCompatibleString().split(' ')[0];
+    var inputNum = inputsObject.toMathjsCompatibleString().split(' ')[0];
 
-    var answerList = this.unitsObjectFactory.fromRawInputString(
-      answerString).toDict();
-    var inputsList = this.unitsObjectFactory.fromRawInputString(
-      inputsString).toDict();
-    return this.utilsService.isEquivalent(answerList, inputsList);
+    return (
+      this.IsEquivalentTo(answer, inputs) &&
```

Comment:
I don't think this is, strictly speaking, correct. For example, 1000 km / L = 1000 m / mL but I wouldn't say these are equal.

I suggest you "normalize" the units -- i.e. for each unit determine its canonical representation (e.g. the canonical representation for 'h', 'hr', 'hrs', 'hours' is 'hr'), convert both the expected and given answers to their canonical representations, then compare them for exact equality. That would satisfy the requirements better.

Reply:
used object to normalize units, now we first normalize the units then compare them.